mindsdb/integrations/handlers/binance_handler/binance_tables.py

```diff
@@ -0,0 +1,175 @@
+from mindsdb.integrations.utilities.date_utils import interval_str_to_duration_ms, utc_date_str_to_timestamp_ms
+from mindsdb.integrations.utilities.sql_utils import extract_comparison_conditions
+from mindsdb_sql.parser import ast
+
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict, List
+
+import pandas as pd
+import time
+
+
+class BinanceAggregatedTradesTable(APITable):
+
+    # Default 1m intervals in aggregate data.
+    DEFAULT_AGGREGATE_TRADE_INTERVAL = '1m'
+    DEFAULT_AGGREGATE_TRADE_LIMIT = 1000
+    # Binance Spot client has connection pool size of 10.
+    MAX_THREAD_POOL_WORKERS = 10
+
+    def _get_batch_klines(self, executor: ThreadPoolExecutor, total_results: int, params: Dict) -> pd.DataFrame:
+        """Gets aggregate trade data in batches and combines the results together.
+        
+        Returns all results as a pandas DataFrame.
+
+        Args:
+            executor (ThreadPoolExecutor): Executor to use when mapping API calls as tasks.
+            total_results (int): Total number of results to fetch.
+            params (Dict): Overall request params to be split into batches.
+        """
+        interval_duration_ms = interval_str_to_duration_ms(params['interval'])
+        if 'end_time' not in params:
+            # Default to get all klines before the current time.
+            overall_end_ms = int(time.time() * 1000)
+        else:
+            overall_end_ms = params['end_time']
+
+        if 'start_time' not in params:
+            total_duration_ms = interval_duration_ms * total_results
+            # Infer start time based on the interval and how many klines we need to fetch.
+            overall_start_ms = overall_end_ms - total_duration_ms
+        else:
+            overall_start_ms = params['start_time']
+
+        next_params = params.copy()
+        next_params['start_time'] = overall_start_ms
+        duration_per_api_call_ms = interval_duration_ms * BinanceAggregatedTradesTable.DEFAULT_AGGREGATE_TRADE_LIMIT
+        next_params['end_time'] = min(overall_end_ms, overall_start_ms + duration_per_api_call_ms)
+        all_params = [next_params.copy()]
+        results_so_far = BinanceAggregatedTradesTable.DEFAULT_AGGREGATE_TRADE_LIMIT
+        while next_params['end_time'] < overall_end_ms and results_so_far < total_results:
+            next_params['limit'] = min(
+                BinanceAggregatedTradesTable.DEFAULT_AGGREGATE_TRADE_LIMIT,
+                total_results - results_so_far
+            )
+            next_params['start_time'] = next_params['end_time']
+            next_params['end_time'] = min(overall_end_ms, next_params['start_time'] + duration_per_api_call_ms)
+            all_params.append(next_params.copy())
+            results_so_far += next_params['limit']
+
+        aggregated_trade_subdatas = list(executor.map(lambda p: self.handler.call_binance_api(method_name='klines', params=p), all_params))
```

Comment:
It sends requests simultaneously, right? This doesn't lead to any block from binance api ?